app/models/manageiq/providers/ibm_cloud/inventory/collector/power_virtual_servers.rb

```diff
@@ -19,14 +19,28 @@ def pvm_instance(instance_id)
     pvm_instances_api.pcloud_pvminstances_get(cloud_instance_id, instance_id)
   end
 
-  def image(img_id)
-    images_api.pcloud_cloudinstances_images_get(cloud_instance_id, img_id)
+  def image(image_id)
+    image = images.detect { |img| img.image_id == image_id }
+    unless image
+      image = images_api.pcloud_cloudinstances_images_get(cloud_instance_id, image_id)
+      images << image
+    end
+    image
   end
 
   def images
     @images ||= images_api.pcloud_cloudinstances_images_getall(cloud_instance_id).images || []
   end
 
+  def image_architecture(image_id)
+    image = image(image_id)
```

Comment:
There are `image_id`s that are not returned by the `GET /images` call (I believe these are copies of "catalog" image and are associated with a single VM and can't be used for provisioning... :shrug:).